Sources/SpeziLocalStorage/LocalStorageEntry.swift

```diff
@@ -0,0 +1,66 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Combine
+import SwiftUI
+
+
+/// Access ``LocalStorage`` entries within a SwiftUI View.
+@propertyWrapper
+public struct LocalStorageEntry<Value>: DynamicProperty { // swiftlint:disable:this file_types_order
+    private let key: LocalStorageKey<Value>
+    
+    @Environment(LocalStorage.self) private var localStorage
+    @State private var internals = LocalStorageEntryInternals<Value>()
+    
+    public var wrappedValue: Value? {
+        get { internals.value }
+        nonmutating set {
+            if let newValue = newValue as? any Equatable, newValue.isEqual(internals.value as Any) {
+                // don't persist the same value again
+            } else {
+                try? localStorage.store(newValue, for: key)
+            }
+        }
+    }
+    
+    public init(_ key: LocalStorageKey<Value>) {
+        self.key = key
+    }
+    
+    public func update() {
+        internals.subscribe(to: key, in: localStorage)
+    }
+}
+
+
+@Observable
+private final class LocalStorageEntryInternals<Value> {
+    fileprivate var value: Value?
+    
+    @ObservationIgnored private var cancellable: AnyCancellable?
```

Comment:
We should probably cancel the observation on deinit of the class here? Would then be triggered if `LocalStorageEntry` is no longer referenced in a view?

Reply:
the docs say that the `AnyCancellable` will auto-cancel itself when it gets deinitialized

Reply:
Sounds good 👍